test/extended/openstack/cloud-provider-config.go

```diff
@@ -66,55 +67,125 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack] The Openshift", f
-						o.Expect(strings.ToLower(usr_property.Value())).To(o.Equal(strings.ToLower(int_property.Value())),
+							propertyName, sectionName, internalConfigMap.name, internalConfigMap.namespace)
+						usrProperty, _ := usrSection.GetKey(propertyName)
+						intProperty, _ := intSection.GetKey(propertyName)
+						o.Expect(strings.ToLower(usrProperty.Value())).To(o.Equal(strings.ToLower(intProperty.Value())),
 							"Unexpected value for property %q on section %q. configMap '%q', namespace '%q'.",
-							property_name, section_name, internalConfigMap.name, internalConfigMap.namespace)
+							propertyName, sectionName, internalConfigMap.name, internalConfigMap.namespace)
 						e2e.Logf("  - Property %q with correct value %q on section %q",
-							property_name, int_property.Value(), section_name)
+							propertyName, intProperty.Value(), sectionName)
 					}
 				}
 			}
 		})
+
+		//https://bugzilla.redhat.com/show_bug.cgi?id=2074471
+		g.It("should update use-octavia and enabled properties depending on the NetworkType", func() {
+
+			userConfigMap := configParams{
+				namespace: "openshift-config",
+				name:      "cloud-provider-config",
+				key:       "config",
+				skip:      nil,
+			}
+
+			ccmConfigMap := configParams{
+				namespace: "openshift-cloud-controller-manager",
+				name:      "cloud-conf",
+				key:       "cloud.conf",
+				skip:      nil,
+			}
+
+			sectionName := "LoadBalancer"
+			propertyNames := []string{"use-octavia", "enabled"}
+
+			cfg, err := e2e.LoadConfig()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			c, err := configv1client.NewForConfig(cfg)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			network, err := c.ConfigV1().Networks().Get(context.Background(), "cluster", metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			userCfg, err := getConfig(clientSet,
+				userConfigMap.namespace,
+				userConfigMap.name,
+				userConfigMap.key)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			ccmCfg, err := getConfig(clientSet,
+				ccmConfigMap.namespace,
+				ccmConfigMap.name,
+				ccmConfigMap.key)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			var ccmPropertyValues []string
+			for _, propertyName := range propertyNames {
+				value, err := getPropertyValue(sectionName, propertyName, ccmCfg)
+				o.Expect(err).NotTo(o.HaveOccurred(), "Error getting the property %q in section %q on configMap "+
+					"%q (namespace: %q)", propertyName, sectionName, ccmConfigMap.name, ccmConfigMap.namespace)
+				ccmPropertyValues = append(ccmPropertyValues, value)
+			}
+
+			g.By(fmt.Sprintf("Checking properties for networkType detected: %q", network.Status.NetworkType))
+			var expectedValues []string
+			if network.Status.NetworkType == "Kuryr" {
+				//Fixed values regardless the content of userCfg.
+				expectedValues = []string{"true", "false"}
+			} else {
+				//Should map the values configured on the userCfg.
+				for _, propertyName := range propertyNames {
```

Comment:
When Kuryr is not used, we fix the value of use-octavia as well, to always be true since lbaasv2 is not supported. Perhaps this test can be updated to take that into account?
https://github.com/openshift/cluster-cloud-controller-manager-operator/blob/master/pkg/cloud/openstack/openstack.go#L85-L93

Reply:
Hello @MaysaMacedo, in such a case the property use-octavia is useless on the cloud-provider-config?

What should happen if use-octavia = false or the property is not present in cloud-provider-config?

Reply:
cluster-cloud-controller-manager (CCCMO) enforces that configuration.
I'm proposing the removal of it from the installer as is not needed https://github.com/openshift/installer/pull/5834

Reply:
Great! Thanks, I will update the test accordingly. No need to check the cloud-provider-config then.

on the eCCM cm, for Kuryr: 

`expectedValues = []string{"true", "false"}`

for any other networkType:

`expectedValues = []string{"true", "#UNDEFINED"}`

Reply:
I think we still need it. If the user changes the user managed configmap, the internal one will continue to enforce the appropriate values.

Reply:
Thanks @MaysaMacedo I updated the name of the test and updated the expected values for NetworkType != Kuryr to fixed values. 

I also tested locally with OpenShiftSDN and Kuryr and in both cases the testcase is passing.

Please take a look and let me know if you think something is missing.